File: Assets/Scripts/TeamBase.js
#pragma strict

var mechSpawnPoint : Transform;
var tankSpawnPoint : Transform;
var jetSpawnPoint : Transform;

var team : int;

// TODO: Until I find a better way to do this...
var tankRedMMIcon : UnityEngine.Texture2D;
var tankBlueMMIcon : UnityEngine.Texture2D;

function SpawnBotTank () {
    Debug.Log("Spawn Tank");

    var pt : Vector3 = tankSpawnPoint.position;
    var r : Quaternion = tankSpawnPoint.rotation;

    var newTank : GameObject;
    newTank = Instantiate(Resources.Load("BotTank2"), pt, r) as GameObject;
    // newTank.GetComponent(TeamPlayer).team = team;
    // TODO: Update team texture...

    // Add a minimap icon for the tank.
    var mmIconTex : Texture2D = (team == 0) ? tankBlueMMIcon : tankRedMMIcon;
    var mmCam : MinimapCamera = Camera.FindObjectOfType(MinimapCamera);
    mmCam.AddMinimapIconTo(newTank.transform, mmIconTex);
}

File: Assets/Scripts/TeamBase.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class TeamBase : MonoBehaviour {
    public Transform mechSpawnPoint;
    public Transform tankSpawnPoint;
    public Transform jetSpawnPoint;

    public int team;

    // TODO: Until I find a better way to do this...
    public Texture2D tankRedMMIcon;
    public Texture2D tankBlueMMIcon;

    public void SpawnBotTank () {
        Debug.Log("Spawn Tank");

        Vector3 pt = tankSpawnPoint.position;
        Quaternion r = tankSpawnPoint.rotation;

        GameObject newTank =
            Instantiate(Resources.Load("BotTank2"), pt, r) as GameObject; // TIDYME: 2018-07-07: Avoid Resources.Load
        // newTank.GetComponent<TeamPlayer>().team = team;
        // TODO: Update team texture...

        // Add a minimap icon for the tank.
        Texture2D mmIconTex = (team == 0) ? tankBlueMMIcon : tankRedMMIcon;
        MinimapCamera.AddMinimapIconTo(newTank.transform, mmIconTex);
    }
}
